### PR TITLE
(324) Consider all relevant Contact states in TotalSummary and Recent Campaigns

### PR DESCRIPTION
**Pull request by theflexbird**
#324
It is dependent from #380.

### CODE REVIEW

File: src/client/api/Campaigns.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const { ApiBase } = require('./ApiBase');
 
 class Campaigns extends ApiBase
 {
-    getCampaigns(customerId)
+    getCampaigns()
```
Comment (kwierchris) [inline src/client/api/Campaigns.js]: This is not an enhancement. The server API as it is currently has to change soon so this parameter was meant to already provide a value that is required in the future.

File: src/client/api/Campaigns.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ class Campaigns extends ApiBase
             then(res => res && res.body).catch(this.getErrorFromResponse);
     }
 
-    getCampaignStats(customerId)
+    getCampaignStats()
```
Comment (kwierchris) [inline src/client/api/Campaigns.js]: This is not an enhancement. The server API as it is currently has to change soon so this parameter was meant to already provide a value that is required in the future.

File: src/client/components/TotalSummaryWidget/TotalSummaryWidget.react.js [hunk 3]
```diff
@@ -41,12 +41,14 @@ class TotalSummaryWidget extends ContextComponent
 
     reload()
     {
-        return this.campaignsApi.getCampaignStats(this.props.customerId).then(stats =>
+        return this.campaignsApi.getCampaignStats().then(stats =>
```
Comment (kwierchris) [inline src/client/components/TotalSummaryWidget/TotalSummaryWidget.react.js]: This is not an enhancement. The server API as it is currently has to change soon so this parameter was meant to already provide a value that is required in the future.

File: src/client/components/TotalSummaryWidget/TotalSummaryWidget.react.js [hunk 4]
```diff
@@ -41,12 +41,14 @@ class TotalSummaryWidget extends ContextComponent
...
         {
-            const results = { };
-            stats.forEach(item => results[item.status] = (results[item.status] || 0) + item.statusCount);
+          const results = {};
+          stats.map( currentValue => TotalSummaryWidget.statuses.map( status => {
```
Comment (kwierchris) [inline src/client/components/TotalSummaryWidget/TotalSummaryWidget.react.js]: Whats the exact reason of making this even more complex?

File: src/server/routes/campaign.js [hunk 5]
```diff
@@ -1,225 +1,250 @@
+const fp = require('lodash/fp');
```
Comment (kwierchris) [inline src/server/routes/campaign.js]: Why do we need even more libraries? I'd really prefer to use as few external libs as possible and to such things with what the language already provides. These new libs add even more black boxes to a code other people should be able understand easily and quickly without having to learn multiple additional external APIs.

File: src/client/api/Campaigns.js (hunk 1 above)
Comment (theflexbird) [reply src/client/api/Campaigns.js]: As you can see the parameter is not used in making a request. And if I remember correctly. The backend can check by session what is customerId.

File: src/client/api/Campaigns.js (hunk 2 above)
Comment (theflexbird) [reply src/client/api/Campaigns.js]: As you can see the parameter is not used in making a request. And if I remember correctly. The backend can check by session what is customerId.

File: src/client/components/TotalSummaryWidget/TotalSummaryWidget.react.js (hunk 4 above)
Comment (theflexbird) [reply src/client/components/TotalSummaryWidget/TotalSummaryWidget.react.js]: Backend change and a data structure is different.

File: src/client/api/Campaigns.js (hunk 1 above)
Comment (kwierchris) [reply src/client/api/Campaigns.js]: We will need this explicitly in the future as this shitty session based stuff is a badly written server side API. I already did in advanced and we'll leave it like that.

File: src/client/api/Campaigns.js (hunk 2 above)
Comment (kwierchris) [reply src/client/api/Campaigns.js]: We will need this explicitly in the future as this shitty session based stuff is a badly written server side API. I already did in advanced and we'll leave it like that.

File: src/server/routes/campaign.js (hunk 5 above)
Comment (theflexbird) [reply src/server/routes/campaign.js]: We are using lodash. I ported the code from frontend to backend. If you want to write your own code for functional programming then be my guest. Just spend some time, please, to do it like guys from lodash team.

**Pull request merged**